src/test/java/org/opensearch/security/dlic/rest/api/UserApiTest.java

```diff
@@ -87,11 +87,17 @@ public void testParallelPutRequests() throws Exception {
         boolean created = false;
         for (HttpResponse response : responses) {
             int sc = response.getStatusCode();
-            if (sc == HttpStatus.SC_CREATED) {
-                Assert.assertFalse(created);
-                created = true;
-            } else {
-                Assert.assertEquals(HttpStatus.SC_CONFLICT, sc);
+            switch (sc) {
+                case HttpStatus.SC_CREATED:
+                    Assert.assertFalse(created);
+                    created = true;
+                    break;
+                case HttpStatus.SC_OK:
+                    Assert.assertTrue(created);
```

Comment:
you can't assert that

Reply:
ok, its because async request won't necessarily return in order right? Then what should I put in this case? It doesn't make sense to put `Assert.assertEquals(HttpStatus.SC_OK, sc);` here since the condition is already `case HttpStatus.SC_OK:`. Or just delete this line?

Reply:
It is because thread pool does not provide a guarantee on task order execution and there is no guarantee of what request is processed first. There is no synchronization/order being enforced. Just delete the assert.